INFOSEC-5268: Updating the Bandwidth fork to match Crowdstrike

ashleybrown7 opened the pull request.

redhatrises committed:
chore: fix typos

redhatrises committed:
Merge pull request #581 from redhatrises/typos

chore: fix typos

redhatrises committed:
chore: prefer host retention policies over hide_host (#582)

* chore: prefer host retention policies over hide_host

* chore: pr review updates

* ci: remove idempotence as we're more interested in first failures

---------

Co-authored-by: Carlos Matos <carlos.matos@crowdstrike.com>

dependabot[bot] committed:
chore(deps): bump ansible/ansible-lint from 24.9.2 to 24.10.0 (#580)

Bumps [ansible/ansible-lint](https://github.com/ansible/ansible-lint) from 24.9.2 to 24.10.0.
- [Release notes](https://github.com/ansible/ansible-lint/releases)
- [Commits](https://github.com/ansible/ansible-lint/compare/v24.9.2...v24.10.0)

---
updated-dependencies:
- dependency-name: ansible/ansible-lint
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

carlosmmatos committed:
fix(falcon_install): fixes issue with deleting sensor when running locally (#584)

* fix(falcon_install): fixes issue with deleting sensor when running localhost

Fixes #578

This PR addresses the issue of prematurely deleting the downloaded
sensor file prior to installation. We have removed the logic to it's own
task file that runs at the end of the play.

* chore: add changelog and some gitignore stuff

carlosmmatos committed:
fix(falcon_configure): fixes issue with unwarranted bool (master image) (#585)

* fix(falcon_configure): fixes issue with unwarranted bool (master image)

Fixes #579

This PR introduces a fix to incorrectly using the |bool filter when
working with prov tokens in master image prep. Also adding the ability
to stop the sensor when working within a master image prep as to prevent
the sensor from potentially updating/restarting via that cloud updates.

* ci: adds testing to ensure provisioning token is set

Prior to this change, we were not checking to see if the prov token was
set which was causing a false positive to occur.

* chore: changelog fragment

* chore: stray from using options dict due to omit specialness

carlosmmatos committed:
fix(falcon_configure): control aid generation wait time and logic

Fixes #583

This PR should address concerns about potential timeouts or "misfires"
around waiting for an AID to be generated on Linux systems (particularly
on Ubuntu 20.04). We've added two ways to help mitigate these issues:
1. Use variables to control the retries/delays. Default to 60 seconds
(6/10).
2. Use a block/rescue so that if the 1st attempt fails, we give it
another try before concluding that something is wrong. We fail with a
message that includes some troubleshooting steps.

carlosmmatos committed:
chore: add changelog fragment

carlosmmatos committed:
fix(falcon_configure): remove block/rescue and up default retry time

After much discussion, this is a safer starting point. We will up the
default amount of time to waiting for an AID to 2 minutes, but will
still allow it to be customizable.

carlosmmatos committed:
fix(falcon_install): improves error message when using falcon_sensor_version_decrement but not enough historical sensors are available (#588)

* fix(falcon_install): fixes issue where failure to download wasn't apparent

Fixes #577

This PR adds better failure message when a user uses
falcon_sensor_version_decrement on newly supported linux OS's. Before,
it would just fail when trying to download a sensor with an incorrect
list index value.

* chore: add changelog fragment

* Update 577-decrement.yml

redhatrises committed:
Merge pull request #586 from carlosmmatos/carlosmmatos/issue583

fix(falcon_configure): control aid generation wait time and logic

carlosmmatos committed:
Bump 4.7.2 (#589)

* chore: add summary and update version

* chore: antsibull changelog artifacts

dependabot[bot] committed:
chore(deps): bump ansible/ansible-lint from 24.10.0 to 25.1.1 (#595)

Bumps [ansible/ansible-lint](https://github.com/ansible/ansible-lint) from 24.10.0 to 25.1.1.
- [Release notes](https://github.com/ansible/ansible-lint/releases)
- [Commits](https://github.com/ansible/ansible-lint/compare/v24.10.0...v25.1.1)

---
updated-dependencies:
- dependency-name: ansible/ansible-lint
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nhinds committed:
Add Falcon Linux Sensor RPM signing GPG key, 2025 (#599)

Older GPG keys have been retained for verifying older RPMs

Commit:
Duplicate "Ansible Community Code of Conduct" Link (#593)

You have included the link to the "Ansible Community Code of Conduct" twice in the "Related Information" section

dependabot[bot] committed:
chore(deps): bump ansible/ansible-lint from 25.1.1 to 25.1.3 (#596)

Bumps [ansible/ansible-lint](https://github.com/ansible/ansible-lint) from 25.1.1 to 25.1.3.
- [Release notes](https://github.com/ansible/ansible-lint/releases)
- [Commits](https://github.com/ansible/ansible-lint/compare/v25.1.1...v25.1.3)

---
updated-dependencies:
- dependency-name: ansible/ansible-lint
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

carlosmmatos committed:
chore: bump v4.7.3 (#600)

carlosmmatos committed:
ci: increasing molecule timeout

sometimes windows stuff takes longer

carlosmmatos committed:
feat: new intel rule modules (#602)

dependabot[bot] committed:
chore(deps): bump ansible/ansible-lint from 25.1.3 to 25.4.0 (#601)

Bumps [ansible/ansible-lint](https://github.com/ansible/ansible-lint) from 25.1.3 to 25.4.0.
- [Release notes](https://github.com/ansible/ansible-lint/releases)
- [Commits](https://github.com/ansible/ansible-lint/compare/v25.1.3...v25.4.0)

---
updated-dependencies:
- dependency-name: ansible/ansible-lint
  dependency-version: 25.4.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

carlosmmatos committed:
Rh collection updates (#603)

carlosmmatos committed:
ci: update requirements.txt for ansible-core version (#604)

Initially we pinned this as there were ssh issues. Need to retest this
to see if they have been fixed

carlosmmatos committed:
ci: invalidate pycache and use requirements.txt for devcontainer (#606)

* ci: invalidate pycache and use requirements.txt for devcontainer

* ci: bring back core to support legacy os'

Since we still support legacy OS's.. for now bring this back, TBD we
will fix this to make sure we can use later versions of ansible core on
older OS's

* ci: update core for ansible test

carlosmmatos committed:
Bump v4.8.0 (#605)

* chore: changelog fragments

* chore: bump v4.8.0

bwappsec commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/infosec-ayu/pr-checks/af4614c8-4312-4ca9-9033-eee57a6fd899?refs=3c1371dc72ab83a4d5e010a2bad68ed9f0675cf1&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/infosec-ayu/pr-checks/af4614c8-4312-4ca9-9033-eee57a6fd899/license?refs=3c1371dc72ab83a4d5e010a2bad68ed9f0675cf1&source=prChecksComment)

:white_check_mark: **code/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/infosec-ayu/project/349faf8a-c9e7-458d-b2ba-81c82a49a97c/pr-check/19f5a789-0c7a-4ced-91d0-a2dcbf111fb7?refs=3c1371dc72ab83a4d5e010a2bad68ed9f0675cf1&source=prChecksComment)

nhempenius reviewed (dismissed):
Gtg

carlosmmatos committed:
fix(falcon_install): remove wildcards from falcon_os_version for Amazon Linux 2 (#608)

- Add specific override task for Amazon Linux 2 to set falcon_os_version
without wildcards
- Maintains existing behavior for all other distributions
- Fixes compatibility issue with Amazon Linux 2 sensor installation
